src/Nerdbank.MessagePack/Converters/HardwareAccelerated.cs

```diff
@@ -0,0 +1,2634 @@
+						Unsafe.Add(ref output, outputOffset) = MessagePackCode.Int32;
+						Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), temp);
+						outputOffset += 5U;
+						break;
+				}
+			}
+
+			return outputOffset;
+		}
+
+		private static nuint WriteLittleEndian(ref byte output, ref long input, nuint inputLength)
+		{
+			nuint outputOffset = 0;
+			for (nuint inputOffset = 0; inputOffset < inputLength; inputOffset++)
+			{
+				long temp = Unsafe.Add(ref input, inputOffset);
+				switch (temp)
+				{
+					case >= MessagePackRange.MinFixNegativeInt:
+						switch (temp)
+						{
+							case <= MessagePackRange.MaxFixPositiveInt:
+								Unsafe.Add(ref output, outputOffset++) = unchecked((byte)(sbyte)temp);
+								break;
+							case <= byte.MaxValue:
+								Unsafe.Add(ref output, outputOffset) = MessagePackCode.UInt8;
+								Unsafe.Add(ref output, outputOffset + 1U) = unchecked((byte)(sbyte)temp);
+								outputOffset += 2U;
+								break;
+							case <= ushort.MaxValue:
+								Unsafe.Add(ref output, outputOffset) = MessagePackCode.UInt16;
+								Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(unchecked((short)temp)));
+								outputOffset += 3U;
+								break;
+							case <= uint.MaxValue:
+								Unsafe.Add(ref output, outputOffset) = MessagePackCode.UInt32;
+								Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(unchecked((int)temp)));
+								outputOffset += 5U;
+								break;
+							default:
+								Unsafe.Add(ref output, outputOffset) = MessagePackCode.UInt64;
+								Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(temp));
+								outputOffset += 9U;
+								break;
+						}
+
+						break;
+					case >= sbyte.MinValue:
+						Unsafe.Add(ref output, outputOffset) = MessagePackCode.Int8;
+						Unsafe.Add(ref output, outputOffset + 1U) = unchecked((byte)(sbyte)temp);
+						outputOffset += 2U;
+						break;
+					case >= short.MinValue:
+						Unsafe.Add(ref output, outputOffset) = MessagePackCode.Int16;
+						Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(unchecked((short)temp)));
+						outputOffset += 3U;
+						break;
+					case >= int.MinValue:
+						Unsafe.Add(ref output, outputOffset) = MessagePackCode.Int32;
+						Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(unchecked((int)temp)));
+						outputOffset += 5U;
+						break;
+					default:
+						Unsafe.Add(ref output, outputOffset) = MessagePackCode.Int64;
+						Unsafe.WriteUnaligned(ref Unsafe.Add(ref output, outputOffset + 1U), BinaryPrimitives.ReverseEndianness(temp));
+						outputOffset += 9U;
+						break;
+				}
+			}
+
+			return outputOffset;
+		}
+
+		private static nuint WriteBigEndian(ref byte output, ref long input, nuint inputLength)
```

Comment:
There seems to be a great deal of code duplicated between the `WriteBigEndian` and `WriteLittleEndian` method groups. In my testing today it seems that just putting a `BitConverter.IsLittleEndian` branch in a single shared method allows much more code reuse and has 0 runtime perf impact because the JIT completely removes the N/A branch of code.
Can we do that here?

Reply:
Ok. I rewrite it for the simpleness.